test/e2e/autoscaling/dns_autoscaling.go

```diff
@@ -57,9 +57,13 @@ var _ = SIGDescribe("DNS horizontal autoscaling", func() {
 	var DNSParams3 DNSParamsLinear
 
 	ginkgo.BeforeEach(func(ctx context.Context) {
-		e2eskipper.SkipUnlessProviderIs("gce", "gke")
 		c = f.ClientSet
 
+		_, err := getDNSReplicas(ctx, c)
+		if err != nil && strings.Contains(err.Error(), "expected 1 DNS deployment") {
+			e2eskipper.Skipf(err.Error())
+		}
```

Comment:
just change the logic to return the error and the replicas and reuse the existing block in line 72

if err != nil {
   if strings.Contains(err.Error(), "expected 1 DNS deployment") && originDNSReplicasCount {
			e2eskipper.Skipf(err.Error())
   }
   framework.Failf("unexpcted error: %v", err)
```

```diff
diff --git a/test/e2e/autoscaling/dns_autoscaling.go b/test/e2e/autoscaling/dns_autoscaling.go
index 98f9945245f..563ba9a2ffe 100644
--- a/test/e2e/autoscaling/dns_autoscaling.go
+++ b/test/e2e/autoscaling/dns_autoscaling.go
@@ -311,7 +311,7 @@ func getDNSReplicas(ctx context.Context, c clientset.Interface) (int, error) {
                return 0, err
        }
        if len(deployments.Items) != 1 {
-               return 0, fmt.Errorf("expected 1 DNS deployment, got %v", len(deployments.Items))
+               return len(deployments.Items), fmt.Errorf("expected 1 DNS deployment, got %v", len(deployments.Items))
        }
 
        deployment := deployments.Items[0]
```

Reply:
Fixed